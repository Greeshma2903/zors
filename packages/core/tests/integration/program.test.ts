import { describe, it } from 'vitest';
import { run } from '../helpers';

describe.concurrent('Program()', async () => {
  it('should print the program/command version', async ({ expect }) => {
    const cli = await run('', 'basic', ['--v']);

    expect(cli.stdout).toMatchSnapshot();
  });

  it('should run the command action', async ({ expect }) => {
    const cli = await run('init', 'basic', ['--commit=true']);

    expect(cli.stdout).toMatchSnapshot();
  });

  it('should run with variadic args', async ({ expect }) => {
    const cli = await run('add', 'basic', [
      'index.html',
      'app.js',
      'styles.css',
    ]);
    expect(cli.stdout).toMatchSnapshot();
  });

  it('should print help output', async ({ expect }) => {
    const cli = await run('', 'basic', ['--h']);
    expect(cli.stdout).toMatchSnapshot();
  });

  it('should validate the input args', async ({ expect }) => {
    try {
      await run('add', 'basic', []);
    } catch (error: any) {
      expect(error.stderr).toMatch(
        'ZorsError: Missing required arg for command `add <...files>`'
      );
    }
  });

  it('should validate the input options', async ({ expect }) => {
    try {
      await run('init', 'basic', []);
    } catch (error: any) {
      expect(error.stderr).toMatch(
        'ZorsError: option `-c, --commit <boolean>` value is missing'
      );
    }
  });

  it('should run with plugins', async ({ expect }) => {
    const cli = await run('init', 'with-plugins', ['--commit=true']);

    expect(cli.stdout).toMatchSnapshot();
  });

  it('should run with tools', async ({ expect }) => {
    const cli = await run('init', 'with-tools', ['--commit=true']);

    expect(cli.stdout).toMatchSnapshot();
  });
});
