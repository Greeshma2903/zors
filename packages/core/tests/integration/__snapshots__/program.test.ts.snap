// Vitest Snapshot v1

exports[`Program() > should print help output 1`] = `
"GIT/v1.0.0

Usage:
  $ git 

Commands:
  init    Initialize an empty git repository
  add     Add files to track
  commit  Creates a commit with a given message
  status  Show git tree status

For more info, run any command with the \`--help\` flag:
  $ git init --help
  $ git add --help
  $ git commit --help
  $ git status --help

Options:
  -h, --help     Display help output 
  -v, --version  Display version number "
`;

exports[`Program() > should print the program/command version 1`] = `"GIT/v1.0.0"`;

exports[`Program() > should run the command action 1`] = `
"Intialized an empty git repository
Created an initial commit"
`;

exports[`Program() > should run with variadic args 1`] = `"Tracking files:  [ 'index.html', 'app.js', 'styles.css' ]"`;
